core/src/main/java/org/apache/gravitino/GravitinoEnv.java

```diff
@@ -120,25 +120,29 @@ public static GravitinoEnv getInstance() {
    *
    * @param config The configuration object to initialize the environment.
    */
-  public void initialize(Config config) {
+  public void initialize(Config config, boolean initGravitinoServerComponent) {
     LOG.info("Initializing Gravitino Environment...");
 
     this.config = config;
     this.metricsSystem = new MetricsSystem();
     metricsSystem.register(new JVMMetricsSource());
 
+    this.eventListenerManager = new EventListenerManager();
+    eventListenerManager.init(
+        config.getConfigsWithPrefix(EventListenerManager.GRAVITINO_EVENT_LISTENER_PREFIX));
+    EventBus eventBus = eventListenerManager.createEventBus();
+
+    if (!initGravitinoServerComponent) {
+      return;
+    }
```

Comment:
The implementation here is quite hacky, it highly relies on where it returns. I think you'd better refactor the current implementation.